src/analyser.cpp

```diff
@@ -2235,59 +2257,66 @@ void Analyser::AnalyserImpl::analyseEquationUnits(const AnalyserEquationAstPtr &
                                                                                            unitsMaps);
 
             if (!isDimensionlessExponent) {
-                auto baseAst = isPower ?
-                                   ast->mPimpl->mOwnedRightChild :
-                                   ast->mPimpl->mOwnedLeftChild;
-                auto exponentUnitsMaps = isPower ?
-                                             rightUnitsMaps :
-                                             unitsMaps;
-                auto exponentUserUnitsMaps = isPower ?
-                                                 rightUserUnitsMaps :
-                                                 userUnitsMaps;
-                auto exponentUnitsMultipliers = isPower ?
-                                                    rightUnitsMultipliers :
-                                                    unitsMultipliers;
-
-                issueDescription = "The unit of " + expression(baseAst) + " is not dimensionless. "
-                                   + expressionUnits(baseAst, exponentUnitsMaps, exponentUserUnitsMaps, exponentUnitsMultipliers) + ".";
+                auto exponentUnitsMaps = isPower ? rightUnitsMaps : unitsMaps;
+                auto exponentUserUnitsMaps = isPower ? rightUserUnitsMaps : userUnitsMaps;
+                auto exponentUnitsMultipliers = isPower ? rightUnitsMultipliers : unitsMultipliers;
+
+                issueDescription = "The unit of " + expression(powerData.mExponentAst) + " is not dimensionless. "
+                                   + expressionUnits(powerData.mExponentAst, exponentUnitsMaps, exponentUserUnitsMaps, exponentUnitsMultipliers) + ".";
             }
         }
 
         // Retrieve the exponent and apply it to our units maps and multipliers.
 
         if (isDimensionlessExponent) {
-            double powerRootValue;
-
             if (isPower) {
-                powerRootValue = powerValue(ast->mPimpl->mOwnedRightChild, noPowerValueOrPowerValueConstant, powerValueAst);
+                powerData.mExponentValue = powerValue(ast->mPimpl->mOwnedRightChild, powerData);
             } else { // AnalyserEquationAst::Type::ROOT.
                 if (ast->mPimpl->mOwnedLeftChild->type() == AnalyserEquationAst::Type::DEGREE) {
                     unitsMaps = rightUnitsMaps;
                     userUnitsMaps = rightUserUnitsMaps;
                     unitsMultipliers = rightUnitsMultipliers;
 
-                    powerRootValue = powerValue(ast->mPimpl->mOwnedLeftChild, noPowerValueOrPowerValueConstant, powerValueAst);
+                    powerData.mExponentValue = powerValue(ast->mPimpl->mOwnedLeftChild, powerData);
                 } else {
                     // No DEGREE element, which means that we are dealing with a
                     // square root.
 
-                    powerRootValue = 2.0;
+                    powerData.mExponentValue = 2.0;
```

Comment:
Can someone save my some thinking and tell my why this is a `2.0` and not a `0.5` please.

Reply:
Because we are dealing with a `root` element when we reach that point and if there is no `degree` element then it means that we want the square root of something, hence the "exponent" value being 2.0.